packages/cli/src/commands/install-cluster-command.ts

```diff
@@ -0,0 +1,245 @@
+import YAML from "yaml";
+import { awsRegions } from "../util/aws-regions";
+import { findPanfactumYaml } from "../util/find-panfactum-yaml";
+import { isEnvironmentConfig } from "../util/is-environment-config";
+import { isPanfactumConfig } from "../util/is-panfactum-config";
+import { printHelpInformation } from "../util/print-help-information";
+import { setupVpc } from "./aws/vpc";
+
+const BOOTSTRAP_GUIDE_URL =
+  "https://panfactum.com/docs/edge/guides/bootstrapping/overview\n";
+
+export class InstallClusterCommand extends Command {
+  static override paths = [["install-cluster"]];
+
+  verbose = Option.Boolean("-v,-verbose", {
+    description: "Enable verbose output",
+  });
+
+  static override usage = Command.Usage({
+    description: "Install a Panfactum cluster",
+    details:
+      "This command sets up a new Panfactum cluster including collecting configuration options and setting up all standard components.",
+    examples: [["Start cluster installation", "pf install-cluster"]],
+  });
+
+  async execute(): Promise<number> {
+    this.context.stdout.write(
+      "Starting Panfactum cluster installation process...\n"
+    );
+
+    // If there's no panfactum.yaml they need to complete the initial setup steps
+    const currentDirectory = process.cwd();
+    const panfactumYamlPath = await findPanfactumYaml(currentDirectory);
+    if (panfactumYamlPath === null) {
+      this.context.stderr.write(
+        pc.red(
+          "ERROR: Could not find panfactum.yaml in the current directory or any parent directory.\n" +
+            "Please ensure you've completed the initial setup steps in the guide here:\n" +
+            BOOTSTRAP_GUIDE_URL
+        )
+      );
+      printHelpInformation(this.context);
+      return 1;
+    }
+
+    // If the environments_dir is not set in the panfactum.yaml file they need to complete the initial setup steps
+    const panfactumYamlContent = await Bun.file(panfactumYamlPath).text();
+    const panfactumConfig: unknown = YAML.parse(panfactumYamlContent);
+    let environmentsDir: string | undefined;
+    if (isPanfactumConfig(panfactumConfig)) {
+      environmentsDir = panfactumConfig.environments_dir;
+    }
+
+    if (typeof environmentsDir !== "string") {
+      this.context.stderr.write(
+        pc.red(
+          "ERROR: environments_dir not defined in panfactum.yaml.\n" +
+            "Please ensure you've completed the initial setup steps in the guide here:\n" +
+            BOOTSTRAP_GUIDE_URL
+        )
+      );
+      printHelpInformation(this.context);
+      return 1;
+    }
+
+    const validRegionsPattern = awsRegions.join("|");
+    const pathRegex = new RegExp(
+      `/${environmentsDir}/([^/]+)/(${validRegionsPattern})(?:/.*)?$`
+    );
+    const match = currentDirectory.match(pathRegex);
+
+    if (!match) {
+      this.context.stderr.write(
+        pc.red(
+          "ERROR: Cluster installation must be run from within a valid region-specific directory.\n" +
```

Comment:
It'd be nice if this wasn't a requirement and the user could select via CLI interaction.

Reply:
Yeah, I'm making the decision to have this instruction in the documentation for running the installer to reduce initial complexity. Happy to add it if we feel it's required at this time around.